src/components/Card.js

```diff
@@ -0,0 +1,268 @@
+import { useCourseData } from "../components/FetchCourses";
+import { Certificate } from "./Certificate";
+import { storage } from "../firebase";
+import { ref, remove } from "firebase/database";
+import { ref as storageRef, deleteObject, listAll } from "firebase/storage";
+import { db } from "../firebase";
+import { useState } from "react";
+
+export const AssignCourseCard = ({ cardTitle, studentList }) => {
+  return (
+    <div className="card w-full bg-base-100 shadow-xl">
+      <div className="card-body items-center text-center">
+        <h2 className="card-title">{cardTitle}</h2>
+        <label className="flex items-center">
+          <input type="checkbox" />
+          <div className="label">
+            <span className="label-text">Student 1</span>
+          </div>
+        </label>
+      </div>
+    </div>
+  );
+};
+
+export const CardWithoutActions = ({ cardContent }) => {
+  return (
+    <div className="card w-full bg-base-100 shadow-md">
+      <div className="card-body">
+        <p>{cardContent}</p>
+      </div>
+    </div>
+  );
+};
+
+export const CourseCards = ({ courseMap }) => {
+  const CourseMaterialsButton = ({ cardCourseID }) => {
+    const { courseMaterialsMap } = useCourseData(cardCourseID);
+
+    return Array.from(courseMaterialsMap.entries()).map(
+      ([key, materialsData]) =>
+        materialsData.courseID === cardCourseID &&
+        (/\.(mp4|mov|avi|mkv)$/i.test(materialsData.fileName) ? (
+          <video key={key} className="video" width="640" height="480" controls>
+            <source src={materialsData.url} type="video/mp4" />
+            Your browser does not support the video tag.
+          </video>
+        ) : (
+          <a
+            key={key}
+            href={materialsData.url}
+            className="btn btn-sm bg-info max-w-xs min-h-10"
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            {materialsData.fileName}
+          </a>
+        ))
+    );
+  };
+
+  return Array.from(courseMap.entries()).map(([courseID, courseData]) => (
```

Comment:
```suggestion
  return Array.from(courseMap.entries()).map(([courseID, courseData]) => (
```

I think carouseMap.entries() + .map() already returns an array, so Array.from seems a bit redundant.

Reply:
Map is an actual map, so my comment is invalid